src/zenml/cli/stack.py

```diff
@@ -176,6 +185,7 @@ def register_stack(
         alerter_name: Name of the alerter for this stack.
         data_validator_name: Name of the data validator for this stack.
         set_stack: Immediately set this stack as active.
+        reset_association: Resets the associations for artifact/metadata stores.
```

Comment:
Let me use the new namings to bring some clarity here. When you register a stack, the corresponding artifact store and the metadata store get coupled. This is done to avoid cases such as an empty artifact store clashing with a populated metadata store (due to caching, the code might end up looking for artifacts that don't exist within the given artifact store) or vice versa. Normally, when you register or update a stack, you need to use either brand new stores or use a combination of stores which was declared before. The flag helps you to bypass this check, where you treat the given stores as a brand new coupling by removing their own old associations.